nrfx/hal/nrf_gpio.h

```diff
@@ -748,8 +756,51 @@ NRF_STATIC_INLINE void nrf_gpio_port_retain_set(NRF_GPIO_Type * p_reg, uint32_t
  * @return Mask of retention domains set, created using @ref nrf_gpio_retain_mask_t.
  */
 NRF_STATIC_INLINE uint32_t nrf_gpio_port_retain_get(NRF_GPIO_Type const * p_reg);
+
+/**
+ * @brief Function for checking the retention setting of a pin.
+ *
+ * @param pin_number Pin number.
+ *
+ * @retval true If pin is retained.
+ * @retval false If pin is not retained.
+ */
+NRF_STATIC_INLINE bool nrf_gpio_pin_retain_check(uint32_t pin_number);
+#endif
+
+#if NRF_GPIO_HAS_RETENTION_SETCLEAR
+/**
+ * @brief Function for setting the retention of the registers.
+ *
+ * @param p_reg Pointer to the structure of registers of the peripheral.
+ * @param mask  Mask of pins to have retention enabled, created using @ref nrf_gpio_retain_mask_t.
+ */
+NRF_STATIC_INLINE void nrf_gpio_port_retain_set(NRF_GPIO_Type * p_reg, uint32_t mask);
+
+/**
+ * @brief Function for clearing the retention of the registers.
+ *
+ * @param p_reg Pointer to the structure of registers of the peripheral.
+ * @param mask  Mask of pins to have retention be disabled, created using @ref nrf_gpio_retain_mask_t.
+ */
+NRF_STATIC_INLINE void nrf_gpio_port_retain_clear(NRF_GPIO_Type * p_reg, uint32_t mask);
```

Comment:
```suggestion
NRF_STATIC_INLINE void nrf_gpio_port_retain_disable(NRF_GPIO_Type * p_reg, uint32_t mask);
```

Reply:
this would be inconsistent with eg `nrf_gpio_port_out_clear`